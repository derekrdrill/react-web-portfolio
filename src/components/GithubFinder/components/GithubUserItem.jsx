import React from 'react';
import styled from 'styled-components';
import { Typography } from '@mui/material';
import { GithubContext } from '../context/GithubContext';
import { getUserAndRepos } from '../context/GithubActions';

export const GithubUserItem = ({ user: { login, avatar_url } }) => {
  const { githubDispatch } = React.useContext(GithubContext);

  const handleGetUser = async () => {
    githubDispatch({ type: 'SET_LOADING' });

    const user = await getUserAndRepos(login);
    githubDispatch({ type: 'GET_USER_AND_REPOS', payload: user });
  };

  return (
    <UserItemContainer>
      <UserImage src={avatar_url} />
      <div>
        <Typography variant='subtitle1' component='h6'>
          {login}
        </Typography>
        <ProfileLink onClick={handleGetUser}>
          <Typography variant='subtitle2' component='h6'>
            View Profile
          </Typography>
        </ProfileLink>
      </div>
    </UserItemContainer>
  );
};

const UserItemContainer = styled.div({
  borderRadius: 10,
  boxShadow: '2px 1px 10px #121326',
  display: 'flex',
  justifyContent: 'space-around',
  margin: 10,
  padding: 5,
});

const UserImage = styled.img({
  borderRadius: 50,
  width: 65,
  height: 65,
});

const ProfileLink = styled.a({
  ':hover': {
    color: '#858585',
    h6: {
      color: '#858585',
    },
  },
  color: '#9c9c9c',
  cursor: 'pointer',
  h6: {
    color: '#9c9c9c',
  },
  textDecoration: 'none',
});
