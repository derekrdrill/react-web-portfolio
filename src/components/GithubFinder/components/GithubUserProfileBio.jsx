import React from 'react';

import { GithubContext } from '../context/GithubContext';

export const GithubUserProfileBio = () => {
  const {
    user: { bio },
  } = React.useContext(GithubContext);

  return bio ? (
    <span className='bio'>{bio}</span>
  ) : (
    <>
      N/A
      <span className='default-bio'>
        'Creative, analytical and passionate React Developer with a demonstrated history of
        delivering effective solutions.'
      </span>
    </>
  );
};
