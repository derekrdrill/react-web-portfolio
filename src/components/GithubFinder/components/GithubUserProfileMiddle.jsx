import React from 'react';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faStore,
  faUsers,
  faUserFriends,
  faProjectDiagram,
} from '@fortawesome/fontawesome-free-solid';

import { GithubContext } from '../context/GithubContext';

export const GithubUserProfileMiddle = () => {
  const {
    user: { followers, following, public_gists, public_repos },
  } = React.useContext(GithubContext);

  const data = [
    { id: 1, title: 'Followers', value: followers, icon: faUsers },
    { id: 2, title: 'Following', value: following, icon: faUserFriends },
    { id: 3, title: 'Public Repos', value: public_repos, icon: faProjectDiagram },
    { id: 4, title: 'Public Gists', value: public_gists, icon: faStore },
  ];

  return (
    <Grid container justifyContent={{ md: 'center', lg: 'flex-start' }}>
      {data.map(item => (
        <MiddleSectionItem key={item.id} item xs={12} md={6} lg={3}>
          <Grid container>
            <Grid item xs={10}>
              <Typography component='h6' variant='subtitle2'>
                {item.title}
              </Typography>
              <Typography component='h5' variant='h5'>
                {item.value}
              </Typography>
            </Grid>
            <Grid item xs={2}>
              <FontAwesomeIcon
                color='lightgrey'
                icon={item.icon}
                style={{ height: '1.5em', width: '1.5em' }}
              />
            </Grid>
          </Grid>
        </MiddleSectionItem>
      ))}
    </Grid>
  );
};

export const MiddleSectionItem = styled(Grid)({
  borderRadius: 10,
  boxShadow: '2px 1px 10px #121326',
  padding: 15,
});
