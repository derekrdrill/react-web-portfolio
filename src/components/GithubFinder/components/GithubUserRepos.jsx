import React from 'react';
import styled from 'styled-components';
import { Chip, Grid, Tooltip, Typography } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';
import { GithubContext } from '../context/GithubContext';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faLink,
  faDatabase,
  faEye,
  faInfo,
  faStar,
  faUtensils,
} from '@fortawesome/fontawesome-free-solid';

export const GithubUserRepos = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);
  const { repos } = React.useContext(GithubContext);

  return (
    <Grid container>
      <GithubReposContainer item xs={12}>
        <TopReposTitleContainer container>
          <TopReposTitle component='h5' variant='h5'>
            Top Repositories
          </TopReposTitle>
        </TopReposTitleContainer>
        {repos.map((repo, repoKey) => {
          const repoInfo = [
            { id: 'size', value: `${repo.size} kb`, color: 'secondary', icon: faDatabase },
            { id: 'watchers', value: repo.watchers, color: 'info', icon: faEye },
            {
              id: 'stargazers_count',
              value: repo.stargazers_count,
              color: 'success',
              icon: faStar,
            },
            { id: 'open_issues', value: repo.open_issues, color: 'error', icon: faInfo },
            { id: 'forks_count', value: repo.forks_count, color: 'warning', icon: faUtensils },
          ];

          return (
            <RepoContainer container key={repoKey} darkMode={darkMode} rowSpacing={2}>
              <CopyLink href={repo.html_url} rel='noreferrer' target='_blank'>
                <CopyLinkIcon icon={faLink} color={darkMode ? 'white' : 'black'} />
              </CopyLink>
              <RepoNameTitle component='h6' variant='subtitle1'>
                {repo.name}
              </RepoNameTitle>
              <Grid item xs={12}>
                <Typography component='h6' variant='subtitle2'>
                  {repo.bio}
                </Typography>
              </Grid>
              <Grid item xs={4}>
                <RepoStatsChipsContainer>
                  {repoInfo.map(item => (
                    <Tooltip key={item.id} title={`${item.id}: ${item.value}`}>
                      <RepoStatsChip
                        label={item.value}
                        color={item.color}
                        icon={<FontAwesomeIcon icon={item.icon} />}
                        variant={darkMode ? 'outlined' : 'filled'}
                        size='small'
                      />
                    </Tooltip>
                  ))}
                </RepoStatsChipsContainer>
              </Grid>
            </RepoContainer>
          );
        })}
      </GithubReposContainer>
    </Grid>
  );
};

const GithubReposContainer = styled(Grid)({
  padding: 35,
});

const RepoContainer = styled(Grid)(({ darkMode }) => ({
  ':hover': {
    backgroundColor: darkMode && '#373975',
    boxShadow: !darkMode && '2px 1px 10px royalblue',
  },
  boxShadow: '2px 1px 10px #121326',
  borderRadius: 10,
  padding: 35,
  marginBottom: 40,
}));

const RepoStatsChip = styled(Chip)({
  '&&.MuiChip-root': {
    margin: 5,
    minWidth: 90,
    padding: '12px 0px 12px 5px',
  },
});

const CopyLink = styled.a({
  marginRight: 10,
});

const CopyLinkIcon = styled(FontAwesomeIcon)({
  marginTop: 8,
});

const TopReposTitleContainer = styled(Grid)({
  marginBottom: 40,
});

const TopReposTitle = styled(Typography)({
  fontWeight: 'bold',
});

const RepoNameTitle = styled(Typography)({
  marginLeft: 5,
});

const RepoStatsChipsContainer = styled.div({
  display: 'flex',
  justifyContent: 'space-between',
});
