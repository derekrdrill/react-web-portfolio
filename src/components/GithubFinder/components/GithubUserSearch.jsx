import React, { useRef, useState, useContext } from 'react';
import styled from 'styled-components';
import { Button, TextField } from '@mui/material';
import { FaTimes } from 'react-icons/fa';
import { AlertComponent as Alert } from '../../Alert/components/AlertComponent';
import { GithubContext } from '../context/GithubContext';
import { AlertContext } from '../../Alert/context/AlertContext';
import { searchUsers } from '../context/GithubActions';

export const GithubUserSearch = () => {
  const searchInput = useRef(null);
  const [text, setText] = useState('');
  const { githubDispatch } = useContext(GithubContext);
  const { alertDispatch } = useContext(AlertContext);

  const handleChange = e => {
    let textValue = e.currentTarget.value;

    if (!textValue) {
      handleClearUsers();
    } else {
      setText(textValue);
    }
  };

  const handleClearUsers = () => {
    githubDispatch({ type: 'CLEAR_USERS' });
    setText('');
    searchInput.current.focus();
  };

  const handleSubmit = async e => {
    e.preventDefault();

    if (text === '') {
      const alertMsg = 'You must enter something for the search to fire ðŸ”¥';
      const alertTitle = 'Empty Search';
      const alertType = 'error';

      alertDispatch({
        type: 'SET_ALERT',
        fadeOut: false,
        payload: { alertMsg, alertTitle, alertType },
      });

      setTimeout(() => alertDispatch({ type: 'FADE_ALERT', fadeOut: true }), 2000);
      setTimeout(() => alertDispatch({ type: 'REMOVE_ALERT' }), 2400);
    } else {
      githubDispatch({ type: 'SET_LOADING' });
      const users = await searchUsers(text);
      githubDispatch({ type: 'GET_USERS', payload: users });
    }
  };

  return (
    <GithubUserSearchContainer>
      <SearchInputContainer>
        <SearchInput
          fullWidth
          label='Type a user name'
          onChange={handleChange}
          inputRef={searchInput}
          value={text}
          variant='outlined'
        />
        {text && <ClearSearch onClick={handleClearUsers} />}
        <SearchButton onClick={handleSubmit} variant='contained'>
          Search
        </SearchButton>
      </SearchInputContainer>
      <AlertContainer>
        <Alert />
      </AlertContainer>
    </GithubUserSearchContainer>
  );
};;;

const GithubUserSearchContainer = styled.div({
  margin: '200px 0px 0px 20px',
});

const SearchInputContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-start',
});

const AlertContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-end',
  marginTop: 10,
  transform: 'translateX(-15px)',
});

const SearchInput = styled(TextField)({
  '&.MuiFormControl-root': {
    ':hover': {
      backgroundColor: '#607080',
    },
    backgroundColor: '#4f5c69',
  },
  '.MuiInputBase-root': {
    borderRadius: 0,
    color: '#75baff',
  },
  '.MuiInputLabel-root': {
    color: '#75baff',
  },
  '.Mui-focused': {
    color: '#bddeff',
    '&.MuiInputBase-root': {
      backgroundColor: '#607080',
    },
  },
});

const SearchButton = styled(Button)({
  borderRadius: 0,
  transform: 'translateX(-15px)',
});

const ClearSearch = styled(FaTimes)({
  ':hover': {
    fill: '#cb1515',
  },
  cursor: 'pointer',
  fill: '#4e0909',
  marginTop: 20,
  transform: 'translateX(-25px)',
});
