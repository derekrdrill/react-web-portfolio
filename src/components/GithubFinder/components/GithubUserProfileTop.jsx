import React from 'react';
import styled from 'styled-components';
import { Button, Chip, Grid, Typography } from '@mui/material';

import { GithubContext } from '../context/GithubContext';

import { GithubUserProfileBio } from './GithubUserProfileBio';

export const getBlog = blog => blog ?? 'N/A';
export const getTwitterUsername = twitter_username => twitter_username ?? 'N/A';

export const GithubUserProfileTop = () => {
  const {
    user: { avatar_url, blog, hireable, html_url, location, login, name, twitter_username, type },
  } = React.useContext(GithubContext);

  return (
    <Grid container spacing={1}>
      <Grid item sm={12} md={3} xl={2}>
        <ProfileImage src={avatar_url} />
        <ProfileNames component='h6' variant='subtitle1'>
          {name}
        </ProfileNames>
        <ProfileNames component='h6' variant='subtitle2'>
          {login}
        </ProfileNames>
      </Grid>
      <Grid item sm={12} md={9} xl={10}>
        <Grid container spacing={1}>
          <Grid item xs={12}>
            <Typography component='h4' variant='h5'>
              {name}
            </Typography>
            <ChipsContainer>
              {type && <Chip label={type} color='success' variant='outlined' size='small' />}
              {hireable && (
                <Chip label='Hireable' color='secondary' variant='outlined' size='small' />
              )}
            </ChipsContainer>
          </Grid>
          <Grid item xs={12}>
            <Typography component='p' variant='subtitle1'>
              <GithubUserProfileBio />
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Button size='large' color='info' variant='outlined' href={html_url} target='_blank'>
              Visit Profile
            </Button>
          </Grid>
          <Grid item xs={12}>
            <Grid container>
              <Grid item xs={12} sm={3}>
                <Typography component='h6' variant='subtitle2'>
                  Location
                </Typography>
                <Typography component='h6' variant='subtitle1'>
                  {location}
                </Typography>
              </Grid>
              <Grid item xs={12} sx={{ display: { xs: 'block', lg: 'none' } }}>
                <hr />
              </Grid>
              <Grid item xs={12} sm={3}>
                <Typography component='h6' variant='subtitle2'>
                  Website
                </Typography>
                <Typography component='h6' variant='subtitle1'>
                  {getBlog(blog)}
                </Typography>
              </Grid>
              <Grid item xs={12} sx={{ display: { xs: 'block', lg: 'none' } }}>
                <hr />
              </Grid>
              <Grid item xs={12} sm={3}>
                <Typography component='h6' variant='subtitle2'>
                  Twitter
                </Typography>
                <Typography component='h6' variant='subtitle1'>
                  {getTwitterUsername(twitter_username)}
                </Typography>
              </Grid>
              <Grid item xs={12} sx={{ display: { xs: 'block', lg: 'none' } }}>
                <hr />
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export const ChipsContainer = styled.div({
  display: 'grid',
  gridGap: 10,
  gridTemplateColumns: '100px 100px',
});

export const ProfileImage = styled.img({
  background: 'black',
  maxHeight: 189,
  maxWidth: 189,
  opacity: 0.7,
});

export const ProfileNames = styled(Typography)({
  color: 'beige',
  transform: 'translate(10px, -50px)',
});
