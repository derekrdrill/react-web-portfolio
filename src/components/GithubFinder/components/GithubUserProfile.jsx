import React, { useContext } from 'react';
import styled from 'styled-components';
import { Button, Chip, Grid, Typography } from '@mui/material';
import { GithubUserRepos } from './GithubUserRepos';
import { GithubContext } from '../context/GithubContext';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStore, faUsers, faUserFriends, faProjectDiagram } from '@fortawesome/fontawesome-free-solid';

const TopSection = ({ avatar_url, bio, blog, hireable, html_url, location, login, name, twitter_username, type }) => (
  <Grid container spacing={1}>
    <Grid item sm={12} md={3} xl={2}>
      <ProfileImage src={avatar_url} />
      <ProfileNames component='h6' variant='subtitle1'>
        {name}
      </ProfileNames>
      <ProfileNames component='h6' variant='subtitle2'>
        {login}
      </ProfileNames>
    </Grid>
    <Grid item sm={12} md={9} xl={10}>
      <Grid container spacing={1}>
        <Grid item xs={12}>
          <Typography component='h4' variant='h5'>
            {name}
          </Typography>
          <ChipsContainer>
            {type && <Chip label={type} color='success' variant='outlined' size='small' />}
            {hireable && <Chip label='Hireable' color='secondary' variant='outlined' size='small' />}
          </ChipsContainer>
        </Grid>
        <Grid item xs={12}>
          <Typography component='p' variant='subtitle1'>
            {bio ? (
              bio
            ) : (
              <>
                N/A
                <span style={{ visibility: 'hidden' }}>
                  'Creative, analytical and passionate React Developer with a demonstrated history of delivering
                  effective solutions.'
                </span>
              </>
            )}
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Button size='large' color='info' variant='outlined' href={html_url} target='_blank'>
            Visit Profile
          </Button>
        </Grid>
        <Grid item xs={12}>
          <Grid container>
            <Grid item xs={12} sm={3}>
              <Typography component='h6' variant='subtitle2'>
                Location
              </Typography>
              <Typography component='h6' variant='subtitle1'>
                {location}
              </Typography>
            </Grid>
            <Grid item xs={12} sx={{ display: { xs: 'block', lg: 'none' } }}>
              <hr />
            </Grid>
            <Grid item xs={12} sm={3}>
              <Typography component='h6' variant='subtitle2'>
                Website
              </Typography>
              <Typography component='h6' variant='subtitle1'>
                {blog ?? 'N/A'}
              </Typography>
            </Grid>
            <Grid item xs={12} sx={{ display: { xs: 'block', lg: 'none' } }}>
              <hr />
            </Grid>
            <Grid item xs={12} sm={3}>
              <Typography component='h6' variant='subtitle2'>
                Twitter
              </Typography>
              <Typography component='h6' variant='subtitle1'>
                {twitter_username ?? 'N/A'}
              </Typography>
            </Grid>
            <Grid item xs={12} sx={{ display: { xs: 'block', lg: 'none' } }}>
              <hr />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  </Grid>
);

const MiddleSection = ({ followers, following, public_gists, public_repos }) => {
  const data = [
    { id: 1, title: 'Followers', value: followers, icon: faUsers },
    { id: 2, title: 'Following', value: following, icon: faUserFriends },
    { id: 3, title: 'Public Repos', value: public_repos, icon: faProjectDiagram },
    { id: 4, title: 'Public Gists', value: public_gists, icon: faStore },
  ];

  return (
    <Grid container justifyContent={{ md: 'center', lg: 'flex-start' }}>
      {data.map(item => (
        <MiddleSectionItem key={item.id} item xs={12} md={6} lg={3}>
          <Grid container>
            <Grid item xs={10}>
              <Typography component='h6' variant='subtitle2'>
                {item.title}
              </Typography>
              <Typography component='h5' variant='h5'>
                {item.value}
              </Typography>
            </Grid>
            <Grid item xs={2}>
              <FontAwesomeIcon color='lightgrey' icon={item.icon} style={{ height: '1.5em', width: '1.5em' }} />
            </Grid>
          </Grid>
        </MiddleSectionItem>
      ))}
    </Grid>
  );
};

export const GithubUserProfile = () => {
  const {
    githubDispatch,
    user: {
      avatar_url,
      bio,
      blog,
      followers,
      following,
      hireable,
      html_url,
      location,
      login,
      name,
      public_gists,
      public_repos,
      twitter_username,
      type,
    },
  } = useContext(GithubContext);

  const handleClearUser = () => {
    githubDispatch({ type: 'SET_LOADING' });
    githubDispatch({ type: 'CLEAR_USER' });
  };

  return (
    <Grid item>
      <BackButtonContainer>
        <Button color='secondary' onClick={handleClearUser}>
          Back to user results
        </Button>
      </BackButtonContainer>
      <MainContainer container>
        <TopSection
          avatar_url={avatar_url}
          bio={bio}
          blog={blog}
          hireable={hireable}
          html_url={html_url}
          location={location}
          login={login}
          name={name}
          twitter_username={twitter_username}
          type={type}
        />
        <MiddleSection
          followers={followers}
          following={following}
          public_gists={public_gists}
          public_repos={public_repos}
        />
        <GithubUserRepos />
      </MainContainer>
    </Grid>
  );
};

const BackButtonContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-start',
});

const MainContainer = styled(Grid)({
  marginTop: 50,
  padding: '0 50px',
});

const ChipsContainer = styled.div({
  display: 'grid',
  gridGap: 10,
  gridTemplateColumns: '100px 100px',
});

const ProfileImage = styled.img({
  background: 'black',
  maxHeight: 189,
  maxWidth: 189,
  opacity: 0.7,
});

const ProfileNames = styled(Typography)({
  color: 'beige',
  transform: 'translate(10px, -50px)',
});

const MiddleSectionItem = styled(Grid)({
  borderRadius: 10,
  boxShadow: '2px 1px 10px #121326',
  padding: 15,
});
