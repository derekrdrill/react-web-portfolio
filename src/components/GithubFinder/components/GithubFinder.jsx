import React from 'react';
import { createGlobalStyle } from 'styled-components';
import { GithubUsersContainer } from './GithubUsersContainer';
import { GithubUserSearch } from './GithubUserSearch';
import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

const GithubFinder = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <GithubUserSearch />
      <GithubUsersContainer />
    </AlertProvider>
  );
};

export default GithubFinder;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#202142' : 'beige',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
