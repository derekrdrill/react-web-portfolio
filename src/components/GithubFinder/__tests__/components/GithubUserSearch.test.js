import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { GithubUserSearch } from '../../components/GithubUserSearch';

const renderer = new ShallowRenderer();

describe('GithubUserSearch tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({ alertDispatch: jest.fn(), githubDispatch: jest.fn() });
    renderer.render(<GithubUserSearch />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
