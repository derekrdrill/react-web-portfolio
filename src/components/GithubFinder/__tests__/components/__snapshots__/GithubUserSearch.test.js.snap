// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubUserSearch tests renders correctly 1`] = `
<styled.div>
  <styled.div>
    <Styled(Component)
      fullWidth={true}
      inputRef={
        Object {
          "current": null,
        }
      }
      label="Type a user name"
      onChange={[Function]}
      value=""
      variant="outlined"
    />
    
    <Styled(Component)
      onClick={[Function]}
      variant="contained"
    >
      Search
    </Styled(Component)>
  </styled.div>
  <styled.div>
    <AlertComponent />
  </styled.div>
</styled.div>
`;
